core/app/finders/concerns/spree/product_filterable.rb

```diff
@@ -3,7 +3,7 @@ module ProductFilterable
     private
 
     def find_available(scope, products_scope)
-      scope.filterable.for_products(products_scope)
+      scope.filterable.order("#{OptionType.table_name}.position, #{OptionValue.table_name}.position").for_products(products_scope)
```

Comment:
@aleksandarpetrushev this concern is also used by properties. You need to move this model-specific code somewhere else

Reply:
@damianlegawiec I just can not get around this error. Even when I select every column. Without .distinct in the :filterable scope of OptionValue it seems to work.